### ISSUE TITLE
When a match arm fits in the max width with {} it is still forced on to multiple lines

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#773]
e.g.

```
         StructLitStyle::Visual => {⏎
-            (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0))⏎
+            (offset + path_str.len() + 3,⏎
+             h_budget.checked_sub(1).unwrap_or(0))⏎
         }⏎
```

Because we try to format it after the `=>`, see it needs two lines and force a `{}`. Even though with the `{}` it will fit on one line

**Comment by Zarokka** [comment]
seems to be fixed in 0.6.2 and current master

this:
```rust
fn main() {
    match x {
        // below is exactly 100 wide and should not be put into {} with line breaks
        StructLitSty::Visual => (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0))
    }

    match x {
        // is the same as above and {} should be removed
        StructLitSty::Visual => {
            (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0))
        }
    }

    match x {
        // below is 102 wide and should be put into {} with line breaks
        StructLitStyle::Visual => (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0))
    }
}
```

outputs:
```rust
fn main() {
    match x {
        // below is exactly 100 wide and should not be put into {} with line breaks
        StructLitSty::Visual => (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0)),
    }

    match x {
        // is the same as above and {} should be removed
        StructLitSty::Visual => (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0)),
    }

    match x {
        // below is 102 wide and should be put into {} with line breaks
        StructLitStyle::Visual => {
            (offset + path_str.len() + 3, h_budget.checked_sub(1).unwrap_or(0))
        }
    }
}
```
which looks alright to me

**Comment by nrc** [comment]
Thanks for checking this!
